### PR TITLE
[Windows] Add scrolling tests for virtualization w/ Datatemplates

### LINKED ISSUE #20926: [Testing, Windows] Add tests to validate DataTemplateSelector WinUI
### Description

Add tests for https://github.com/dotnet/maui/pull/20875

I merged the PR as is because I wanted to get the fix out on our nightly feed and we have enough time right now to add some additional tests before going stable.

### PR DESCRIPTION
**Pull request by Foda**
### Description of Change

Add scrolling tests for virtualization w/ Datatemplates

### Issues Fixed

Fixes #20926

### CODE REVIEW

File: src/Controls/tests/UITests/Tests/Issues/Issue20842.cs [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+			App.WaitForElement(scrollUpButton);
+
+			App.Click(scrollDownButton);
+			await Task.Delay(200);
+			App.Click(scrollUpButton);
+			await Task.Delay(200);
+			App.Click(scrollDownButton);
+
+			VerifyScreenshot();
```
Comment (jsuarezruiz) [inline src/Controls/tests/UITests/Tests/Issues/Issue20842.cs:31]: Snapshot different than baseline: VerifyCollectionViewItemsAfterScrolling.png (14.21% difference)
image

**Review by jsuarezruiz — COMMENTED** [review]
Seems that there are differences between the added snapshot and the image generated in CI server.

File: src/Controls/tests/UITests/Tests/Issues/Issue20842.cs (hunk 1 above)
Comment (Foda) [reply src/Controls/tests/UITests/Tests/Issues/Issue20842.cs:32]: Looks like I forgot to add another delay to allow the items to hydrate in...

**Review by PureWeen — APPROVED** [review]

**Pull request merged**